From cfd41bc70073f9dcebce6c290324115098c7d81f Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Thu, 19 Nov 2020 17:54:48 +0200
Subject: [PATCH] [SW-14663] make: add -fno-builtin into CFLAGS

Add -fno-builtin in CFLAGS so that no builtin
are added into code. This is done in order to
prevent unwanted implementation from compiler
for example: memcpy, memcmp, etc.

Change-Id: Ic5143a79ebf11a2c87849bdb8afba0433775cae2
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/Makefile b/Makefile
index 620d405..dadb547 100644
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
 CFLAGS += -include $(SPECS_EMBEDDED_DIR)/$(SPECS_DEV)/$(SPECS_DEV)_reg_map.h
 CFLAGS += -Wall -Werror
-CFLAGS += -g -c
+CFLAGS += -g -c -fno-builtin
 CFLAGS += -I ./include/ -I ./ -I ./common/ -I ./$(HL_DEV)/
 
 ifeq ($(HL_TARGET),pldm)
